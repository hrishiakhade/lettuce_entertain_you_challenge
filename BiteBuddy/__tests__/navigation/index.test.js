import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import 'react-native-gesture-handler';
import renderer from 'react-test-renderer';
import Navigation from '../../navigation';

jest.mock('@react-navigation/native', () => ({
    NavigationContainer: ({ children }) => <>{children}</>,
}));

jest.mock('@react-navigation/stack', () => ({
    createStackNavigator: () => ({
        Navigator: ({ children }) => <>{children}</>,
        Screen: ({ children }) => <>{children}</>,
    }),
}));

describe('Navigation component', () => {
    test('renders correctly', () => {
        const component = renderer.create(<Navigation />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });


    // renders HomeScreen and ResultScreen correctly
    test('renders Navigation component correctly', () => {
        const component = renderer.create(
            <NavigationContainer>
                <Navigation />
            </NavigationContainer>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
    
});
