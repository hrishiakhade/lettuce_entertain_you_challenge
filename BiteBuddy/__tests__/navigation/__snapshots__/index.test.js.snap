// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders Navigation component correctly 1`] = `null`;

exports[`Navigation component renders correctly 1`] = `null`;
